src/Titon/Console/Command/AbstractCommand.hh

```diff
@@ -0,0 +1,377 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Console\Command;
+
+use Titon\Console\Feedback\ProgressBarFeedback;
+use Titon\Console\Feedback\WaitFeedback;
+use Titon\Console\InputBag;
+use Titon\Console\Command;
+use Titon\Console\Input;
+use Titon\Console\Output;
+use Titon\Console\InputDefinition\Argument;
+use Titon\Console\InputDefinition\Flag;
+use Titon\Console\InputDefinition\Option;
+use Titon\Console\UserInput\Confirm;
+use Titon\Console\UserInput\Menu;
+use Titon\Console\UserInput\Prompt;
+
+/**
+ * A `Command` is a class that configures necessary command line inputs from the
+ * user and executes its `run` method when called.
+ *
+ * @package Titon\Console\Command
+ */
+abstract class AbstractCommand implements Command {
+
+    /**
+     * Bag container holding all registered `Argument` objects
+     *
+     * @var \Titon\Console\InputBag<Argument>
+     */
+    protected InputBag<Argument> $arguments;
+
+    /**
+     * The description of the command used when rendering its help screen.
+     *
+     * @var string
+     */
+    protected string $description = '';
+
+    /**
+     * Bag container holding all registered `Flag` objects
+     *
+     * @var \Titon\Console\InputBag<Flag>
+     */
+    protected InputBag<Flag> $flags;
+
+    /**
+     * The `Input` object containing all registered and parsed command line
+     * parameters.
+     *
+     * @var \Titon\Console\Input
+     */
+    protected Input $input;
+
+    /**
+     * The name of the command passed into the command line.
+     *
+     * @var string
+     */
+    protected string $name = '';
+
+    /**
+     * Bag container holding all registered `Option` objects
+     *
+     * @var \Titon\Console\InputBag<Option>
+     */
+    protected InputBag<Option> $options;
+
+    /**
+     * The `Output` object to handle output to the user.
+     *
+     * @var \Titon\Console\Output
+     */
+    protected Output $output;
+
+    /**
+     * Construct a new instance of a command.
+     *
+     * @param \Titon\Console\Input|null $input  The `Input` object to inject
+     * @param \Titon\Console\Output|null $input The `Output` object to inject
+     */
+    public function __construct(?Input $input = null, ?Output $output = null) {
```

Comment:
Perhaps this input and output is no longer needed as they are explicitly set from the `bootstrap()` call.